mlxtend/frequent_patterns/tests/test_fpbase.py

```diff
@@ -69,13 +69,21 @@ def test_itemsets_type(self):
             assert isinstance(i, frozenset) is True
 
     def test_raise_error_if_input_is_not_binary(self):
+        def test_with_dataframe(df):
+            assert_raises(ValueError,
+                          'The allowed values for a DataFrame are True, '
+                          'False, 0, 1. Found value 2',
+                          self.fpalgo, df)
         df2 = pd.DataFrame(self.one_ary, columns=self.cols).copy()
         df2.iloc[3, 3] = 2
-
-        assert_raises(ValueError,
-                      'The allowed values for a DataFrame are True, '
-                      'False, 0, 1. Found value 2',
-                      self.fpalgo, df2)
+        test_with_dataframe(df2)
+        sdf = df2.to_sparse()
+        test_with_dataframe(sdf)
+        try:
+            sdf = df2.astype(pd.SparseDtype(int, fill_value=0))
+        except AttributeError:
```

Comment:
Same here regarding the ` from distutils.version import LooseVersion as Version` approach mentioned above

Reply:
Will do the requested changes; and here I forgot to call `test_with_dataframe(sdf)` :-/